src/Material.cpp

```diff
@@ -663,38 +435,400 @@ void Material::setNumEnergyGroups(const int num_groups) {
+                                             
+  return scatter_source;
+    
+}
+
+
+/**
+ * @brief Get the Material's diffustion coefficient for some energy group.
+ * @param group the energy group
+ * @return the diffusion coefficient
+ */
+FP_PRECISION MacroMaterial::getDifCoefByGroup(int group) {
+  if (_dif_coef == NULL)
+    log_printf(ERROR, "Unable to return Material %d's diffusion coefficient "
+               "since it has not yet been set", _id);
+  
+  if (group <= 0 || group > _num_groups)
+    log_printf(ERROR, "Unable to get dif_coef for group %d for Material "
+               "%d which contains %d energy groups", group, _id, _num_groups);
+  
+  return _dif_coef[group-1];
+}
+
+
+/**
+ * @brief Get the Material's dif_hat for some energy group.
+ * @param group the energy group
+ * @param surface the index of the surface
+ * @return the dif_hat value
+ */
+FP_PRECISION MacroMaterial::getDifHatByGroup(int group, int surface) {
+  if (_dif_hat == NULL)
+    log_printf(ERROR, "Unable to return Material %d's dif_hat "
+               "since it has not yet been set", _id);
+  
+  if (group <= 0 || group > _num_groups)
+    log_printf(ERROR, "Unable to get dif_hat for group %d for Material "
+               "%d which contains %d energy groups", group, _id, _num_groups);
+               
+  return _dif_hat[surface*_num_groups + (group-1)];
+}
+
+
+/**
+ * @brief Get the Material's dif_tilde for some energy group.
+ * @param group the energy group
+ * @return the dif_tilde value
+ */
+FP_PRECISION MacroMaterial::getDifTildeByGroup(int group) {
+  if (_dif_tilde == NULL)
+    log_printf(ERROR, "Unable to return Material %d's dif_tilde "
+               "since it has not yet been set", _id);
+  
+  if (group <= 0 || group > _num_groups)
+    log_printf(ERROR, "Unable to get dif_tilde for group %d for Material "
+               "%d which contains %d energy groups", group, _id, _num_groups);
+  
+  return _dif_tilde[group-1];
+}
+
+
+/**
+ * @brief Get the Material's buckling for some energy group.
+ * @param group the energy group
+ * @return the buckling value
+ */
+FP_PRECISION MacroMaterial::getBucklingByGroup(int group) {
+  if (_buckling == NULL)
+    log_printf(ERROR, "Unable to return Material %d's buckling "
+               "since it has not yet been set", _id);
+  
+  if (group <= 0 || group > _num_groups)
+    log_printf(ERROR, "Unable to get buckling for group %d for Material "
+               "%d which contains %d energy groups", group, _id, _num_groups);
+  
+  return _buckling[group-1];
+}
+
+
+/**
+ * @brief Set the number of energy groups for this Material.
+ * @param num_groups the number of energy groups.
+ */
+void MacroMaterial::setNumEnergyGroups(const int num_groups) {
+  Material::setNumEnergyGroups(num_groups);
+  
+  /* Free old data arrays if they were allocated for a previous simulation */
+
+  /* If data is vector aligned */
+  if (_data_aligned) {
+    if (_sigma_s != NULL)
+      MM_FREE(_sigma_s);
```

Comment:
You only need to use ```MM_FREE`` if the data is aligned (*e.g.*, within the ``if(_data_aligned) { ... }`` block. Otherwise, it is fine to used ``delete []``. The constructor has to be able to handle both cases - aligned or not - hence the two different blocks here.